packages/open-payments/src/client/requests.ts

```diff
@@ -214,3 +214,30 @@ export const createAxiosInstance = (args: {
 
   return axiosInstance
 }
+
+export type InterceptorFn = (
+  config: InternalAxiosRequestConfig
+) => InternalAxiosRequestConfig | Promise<InternalAxiosRequestConfig>
+
+export const createCustomAxiosInstance = (args: {
+  requestTimeoutMs: number
+  requestInterceptor: InterceptorFn
+}): AxiosInstance => {
+  const axiosInstance = axios.create({
+    headers: {
+      common: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      }
+    },
+    timeout: args.requestTimeoutMs
+  })
+
+  axiosInstance.interceptors.request.use(args.requestInterceptor, undefined, {
+    runWhen: (config: InternalAxiosRequestConfig) =>
+      config.method?.toLowerCase() === 'post' ||
+      !!(config.headers && config.headers['Authorization'])
```

Comment:
I think because we are specifying POST with Authorization `runWhen` here (and not leaving it up to the caller), we should probably make the `requestInterceptor` name more explicit? like `authenticatedRequestInterceptor` or something along those lines.